### PR TITLE
Adding navbar component

### PR DESCRIPTION
**Pull request by lzboy14**
Create Navbar Component

https://www.notion.so/uwblueprintexecs/e2d9163a2f8c436bbb36172066753d40?v=9518aa1af04b4478b3d5e840f4ffcc7f&p=1036e39e516b4c4db2cea6842bc1c2eb&pm=s

Added Navbar component. Also created dummy components for Dashboard, Donation History, Account Management, and the Donate button page. Created Layout.tsx component for storage of the navbar - will need to be implemented to the appropriate page later. The Don Efficace logo button routes to the home page ("/").

* indicates a completely new file/folder

- frontend
    - src
        - components
            - temp_navbar *
                - AccountManagement.tsx *
                - Dashboard.tsx *
                - Donate.tsx *
                - DonationHistory.tsx *
            -   common
                - Layout.tsx *
                - NavBar.tsx *
            - constants
                - Routes.ts (modified - added temporary routes for donate and layout)
            - App.tsx (modified - added route for donate button and Layout.tsx)

1. Play around with the navbar.
2. Drag window size to ensure the navbar is responsive.
3. Try clicking each of the tabs, including back and forth. Ensure that they are properly routing to the correct locations.

Let me know if there are any changes that you think I should make.

## Checklist
- [y] My PR name is descriptive and in imperative tense
- [y] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [y] I have run the appropriate linter(s)
- [y] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -106,6 +106,37 @@ Open Prisma studio:
...
+Kill the occupying service:
+```bash
+sudo kill <PID>
+```
+
+Run the application again:
+```bash
+docker compose up --build
+```
+
```
Comment (mat-ng) [inline README.md:139]: I think you accidentally duplicated your commits when doing the ngrok work. Can you look into removing them from this branch?

File: backend/typescript/Dockerfile [hunk 2]
```diff
@@ -9,14 +9,20 @@ COPY package.json yarn.lock tsconfig.json ./
 
 RUN echo "deb http://archive.debian.org/debian stretch main" > /etc/apt/sources.list
 
-# libcurl3 is required for mongodb-memory-server, which is used for testing
-RUN apt-get update && apt-get install -y libcurl3
+# Install dependencies
```
Comment (mat-ng) [inline backend/typescript/Dockerfile:12]: Same comment about duplicated commits applies here.

File: frontend/src/components/auth/Login.tsx [hunk 3]
```diff
@@ -44,6 +45,13 @@ const Login = (): React.ReactElement => {
     return <Redirect to={HOME_PAGE} />;
   }
 
+  // For testing purposes
+  const filter = {
+    causes: [], // empty array will include all causes
+    frequencies: ["Monthly", "One-time"], // if not empty, only donations with these frequencies will be included
+    years: [],
+  };
+
```
Comment (mat-ng) [inline frontend/src/components/auth/Login.tsx:54]: You might have also copied some extra commits from Jordan and Patrick's branch too, it seems. Can you look into removing these too?

File: frontend/src/index.css [hunk 4]
```diff
@@ -1,8 +1,14 @@
+@import url('https://fonts.googleapis.com/css2?family=Nunito+Sans:ital,opsz,wght@0,6..12,200..1000;1,6..12,200..1000&display=swap');
+
+* {
+  font-family: "Nunito Sans", serif;
+}
+
 body {
   margin: 0;
-  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
+  /* font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
```
Comment (mat-ng) [inline frontend/src/index.css:9]: Why are we commenting this out?

**Review by mat-ng — COMMENTED** [review]
The navbar looked _really_ good when you showed it in the demo, nice work Leo!

It just looks like you pushed some extra commits to this branch, which will cause some merge conflicts when you try to merge this branch into main. If you could just look into removing those changes, that'd be great! Let me know if you have any questions! 🙏

File: README.md (hunk 1 above)
Comment (lzboy14) [reply README.md:139]: Got it. I'm not sure why this is here. I will delete it.

File: frontend/src/index.css (hunk 4 above)
Comment (lzboy14) [reply frontend/src/index.css:9]: I think this may have been from Patrick and Jordan's commit, sorry.

File: frontend/Dockerfile [hunk 5]
```diff
@@ -6,6 +6,12 @@ WORKDIR /app
 COPY package.json yarn.lock ./
 RUN yarn install
 
+# Install ngrok using curl
+RUN curl -o ngrok-stable-linux-amd64.zip https://bin.equinox.io/c/4VmDzA7iaHb/ngrok-stable-linux-amd64.zip && \
+    unzip ngrok-stable-linux-amd64.zip && \
+    mv ngrok /usr/local/bin/ && \
+    rm ngrok-stable-linux-amd64.zip
```
Comment (mat-ng) [inline frontend/Dockerfile:13]: uncommit

File: backend/typescript/swagger.yml [hunk 6]
```diff
@@ -806,7 +806,7 @@ paths:
         '401':
           description: Unauthorized
         '500':
-          description: Invalid request
+          description: Error creating payment checkout session.
```
Comment (mat-ng) [inline backend/typescript/swagger.yml:809]: uncommit

**Review by mat-ng — COMMENTED** [review]
uncommit these changes

**Review by mat-ng — APPROVED** [review]

**Pull request merged**